src/python_testing/TC_BRBINFO_4_1.py

```diff
@@ -0,0 +1,287 @@
+
+        logging.info("Commissioning of ICD to fabric two (DUT)")
+        params = await self.openCommissioningWindow(dev_ctrl=self.default_controller, node_id=self.icd_nodeid)
+
+        self._ask_for_vendor_commissioniong_ux_operation(params.randomDiscriminator, params.commissioningParameters.setupPinCode)
+
+    def teardown_class(self):
+        logging.warning("Stopping app with SIGTERM")
+        self.app_process.send_signal(signal.SIGTERM.value)
+        test_app_exit_code = self.app_process.wait()
+        os.remove(self.kvs)
+        super().teardown_class()
+
+    #
+    # BRBINFO 4.1 Test Body
+    #
+
+    @async_test_body
+    async def test_TC_BRBINFO_4_1(self):
+
+        icdmCluster = Clusters.Objects.IcdManagement
+        icdmAttributes = icdmCluster.Attributes
+        brbinfoCluster = Clusters.Objects.BridgedDeviceBasicInformation
+        brbinfoAttributes = brbinfoCluster.Attributes
+        basicinfoCluster = Clusters.Objects.BasicInformation
+        basicinfoAttributes = basicinfoCluster.Attributes
+
+        self.endpoint = 0
+
+        # Check commissioned DUT
+        self.step(1)
+        # Confirms commissioning of DUT as it reads basic info from bridge device
+        featureMap = await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basicinfoCluster,
+          basicinfoAttributes.FeatureMap,
+          self.dut_node_id
+        )
+
+        self.step(2)
+        # Confirms commissioning of ICD as it reads basic info from ICD device
+        reachable = await self._read_attribute_expect_success(
+          kRootEndpointId,
+          basicinfoCluster,
+          basicinfoAttributes.FeatureMap,
+          self.icd_nodeid
+        )
+
+        self.step(3)
+        # Test 1A
+        idle_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdmCluster,
+          icdmAttributes.IdleModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"IdleModeDuration: {idle_mode_duration}")
+
+        active_mode_duration = await self._read_attribute_expect_success(
+          kICDMEndpointId,
+          icdmCluster,
+          icdmAttributes.ActiveModeDuration,
+          self.icd_nodeid
+        )
+        logging.info(f"ActiveModeDuration: {active_mode_duration}")
+
+        self.step(4)
+        # Test 1B
+
+        # Subscription to ActiveChanged
+        event = brbinfoCluster.Events.ActiveChanged
+        self.q = queue.Queue()
+        urgent = 1
+        cb = SimpleEventCallback("ActiveChanged", event.cluster_id, event.event_id, self.q)
+        subscription = await self.default_controller.ReadEvent(nodeid=self.dut_node_id, events=[(kICDBridgedEndpointId, event, urgent)], reportInterval=[1, 3])
+        subscription.SetEventUpdateCallback(callback=cb)
+
+        StayActiveDuration = 1000
```

Comment:
local variable names should use lower_snake_case

Reply:
ack